packages/registrar_client/src/modules/node/service.ts

```diff
@@ -0,0 +1,43 @@
+import { RegistrarClient } from "../../client/client";
+import { NodeRegistrationRequest, NodesFilter, UpdateNodeRequest, UptimeReportRequest, Node, NodeRegistrationResponse } from "./types";
+import { createAuthHeader } from "../../utils";
+
+export class Nodes {
+  private client: RegistrarClient;
+  private readonly nodeUri = "/nodes";
+
+  constructor(client: RegistrarClient) {
+    this.client = client;
+  }
+
+  async registerNode(node: NodeRegistrationRequest): Promise<NodeRegistrationResponse> {
```

Comment:
Should we add any validation to the node data? Not sure if it's correct to rely on the server validation